python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -165,20 +167,36 @@ def parseAndRun(cls, *args, **kwargs):
             pickle.dump(resultList, open(results.parsedCmd.dumpRatios, "wb"))
         coeff, coeffErr, coeffNum = task.reduce(resultList)
 
+        calib = CrosstalkCalib()
+        provenance = IsrProvenance()
+
+        calib.coeffs = coeff
+        calib.coeffErr = coeffErr
+        calib.coeffNum = coeffNum
+
         outputFileName = results.parsedCmd.outputFileName
         if outputFileName is not None:
             butler = results.parsedCmd.butler
             dataId = results.parsedCmd.id.idList[0]
-            dataId["detector"] = butler.queryMetadata("raw", ["detector"], dataId)[0]
 
+            # Rework to use new crosstalk calib.
             det = butler.get('raw', dataId).getDetector()
-            writeCrosstalkCoeffs(outputFileName, coeff, det=det,
-                                 crosstalkName=results.parsedCmd.crosstalkName, indent=2)
+            calib._detectorName = det.getName()
+            calib._detectorSerial = det.getSerial()
+            calib.nAmp = len(det)
+            calib.hasCrosstalk = True
+            calib.writeText(outputFileName + ".yaml")
+
+            provenance.calibType = 'CROSSTALK'
+            provenance._detectorName = det.getName()
+            provenance.fromDataIds(results.parsedCmd.id.idList)
+            provenance.writeText(outputFileName + '_prov.yaml')
 
         return Struct(
             coeff=coeff,
             coeffErr=coeffErr,
-            coeffNum=coeffNum
+            coeffNum=coeffNum,
+            calib=calib,
```

Comment:
Please add this returned value to the docstring, and while you're at it, maybe make the summary something moderately more informative than "implement scatter/gather" because I don't know what that means.

Reply:
Done.  I've also noted that this method will disappear in DM-24760, as the current logic is confusing at best.